packages/mdc-textfield/foundation.js

```diff
@@ -105,9 +106,10 @@ class MDCTextFieldFoundation extends MDCFoundation {
    */
   constructor(adapter, foundationMap = /** @type {!FoundationMapType} */ ({})) {
     super(Object.assign(MDCTextFieldFoundation.defaultAdapter, adapter));
-
     /** @type {!MDCTextFieldHelperTextFoundation|undefined} */
     this.helperText_ = foundationMap.helperText;
+    /** @type {!MDCFloatingLabelFoundation|undefined} */
```

Comment:
I'm not sure there's a need to add this to the foundation. The foundation itself does not depend on `setLabelContent` anywhere, so it can go directly in the component, in which case it can simply reference `label_` on the component instance (which already exists).